R/spliceHelperFunctions.R

```diff
@@ -479,43 +487,41 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
   return(relist(unlistedExons, partitioning))
 }
 
-#' Function that selects the first exon from an IRangesList object (exon_rank is required)
-#' @param range IRangesList
+#' Function that selects the first exon from an IRangesList object
+#' @param range IRangesList with elements required to be ordered by coordinates
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
   largestExons <- as.numeric(cumsum(elementNROWS(range)))
-  startExonsSet <- c(1, largestExons[1:length(largestExons)-1]+1)
+  startExonsSet <- c(1, largestExons[-(length(largestExons))]+1)
   startExonsSet[strand == "-"] <- largestExons[strand == "-"]
-  return(relist(unlist(range, use.names = FALSE)[startExonsSet],partitioning))
+  return(unlist(range, use.names = FALSE)[startExonsSet])
 }
 
-#' Function that selects the last exon from an IRangesList object (exon_rank is required)
-#' @param range IRangesList
+#' Function that selects the last exon from an IRangesList object
+#' @param range IRangesList with elements required to be ordered by coordinates
 #' @param stand strand
 #' @noRd
 selectEndExonFromRangesList <- function(range, strand){
-  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
   endExonsSet <- as.numeric(cumsum(elementNROWS(range)))
-  smallestExons <- c(1, endExonsSet[1:length(endExonsSet)-1]+1)
+  smallestExons <- c(1, endExonsSet[-(length(endExonsSet))]+1)
```

Comment:
firstExons instead of smallestExons? (rank != size)

Reply:
"smallestExons" is renamed to "firstExons".